fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/utils/FedoraTypesUtils.java

```diff
@@ -122,6 +123,36 @@
     public static Predicate<Node> isInternalNode = uncheck(n -> n.isNodeType("mode:system"));
 
     /**
+     * Check if a node is externally managed.
+     *
+     * Note: modeshape uses a source-workspace-identifier scheme
+     * to identify whether a node is externally-managed.
+     * Ordinary (non-external) nodes will have simple UUIDs
+     * as an identifier. These are never external nodes.
+     *
+     * External nodes will have a 7-character hex code
+     * identifying the "source", followed by another
+     * 7-character hex code identifying the "workspace", followed
+     * by a "/" and then the rest of the "identifier".
+     *
+     * Following that scheme, if a node's "source" key does not
+     * match the repository's configured store name, then it is an
+     * external node.
+     */
+    public static Predicate<Node> isExternalNode = uncheck(n ->  {
+        if (NodeKey.isValidRandomIdentifier(n.getIdentifier())) {
+            return false;
+        } else {
+            final NodeKey key = new NodeKey(n.getIdentifier());
+            final String source = NodeKey.keyForSourceName(
+                    ((JcrRepository)n.getSession().getRepository())
+                        .getConfiguration()
+                        .getStoreName());
+            return key.getSourceKey() != source;
```

Comment:
My mistake, it should be the latter. More importantly, though, there should be a unit test for this.